app/services/transfer_product_from_stock_service.rb

```diff
@@ -0,0 +1,17 @@
+class TransferProductFromStockService
+  def initialize(product:, location:, second_location:, quantity:)
```

Comment:
Can we rename `location` to `source_location` and `second_location` to `destination_location`? To make the names more meaningful. Also in controller and views.